app/src/main/java/com/samco/trackandgraph/antlr/ast/Model.kt

```diff
@@ -0,0 +1,167 @@
+//   Copyright 2021 Federico Tomassetti
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+//  Modified
+
+package com.samco.trackandgraph.antlr.ast
+
+import java.util.*
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.full.primaryConstructor
+import org.antlr.v4.runtime.tree.ParseTree
+
+//
+// Generic part: valid for all languages
+//
+
+interface Node {
```

Comment:
Pretty much the same as for the mapping. Mostly the original code, not much to change since it works great